[4.2rc1] test_bbox_size failure with Python 3.8/3.9

### Description

When I run the test suite during the build of the Debian package, `test_bbox_size` fails on Python 3.8 and 3.9.

### Expected behavior

The tests should succeed on all supported Python versions.

### Actual behavior

I get the following failure:
```
________________________________ test_bbox_size ________________________________

    def test_bbox_size():
        # Test for the size of a WCSAxes bbox
        x0 = 11.38888888888889
        y0 = 3.5
        # Upper bounding box limits (only have Matplotlib >= 3.0 now)
        x1 = 576.0
        y1 = 432.0
    
        fig = plt.figure()
        ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8])
        fig.add_axes(ax)
        fig.canvas.draw()
        renderer = fig.canvas.renderer
        ax_bbox = ax.get_tightbbox(renderer)
>       assert np.allclose(ax_bbox.x0, x0)
E       assert False
E        +  where False = <function allclose at 0x7fb26d2d40d0>(11.26388888888889, 11.38888888888889)
E        +    where <function allclose at 0x7fb26d2d40d0> = np.allclose
E        +    and   11.26388888888889 = Bbox([[11.26388888888889, 3.5], [576.0, 432.0]]).x0

astropy/visualization/wcsaxes/tests/test_misc.py:521: AssertionError
= 1 failed, 15010 passed, 294 skipped, 60 xfailed, 12 xpassed in 717.27 seconds =
```

### Steps to Reproduce

Build (and test) the Debian experimental package (soon [in salsa](https://salsa.debian.org/debian-astro-team/astropy/-/tree/experimental)).

### System Details

 * Python 3.8.6, 3.9.0
 * Numpy 1.19.4
 * Astropy 4.2rc1
 * Scipy 1.5.3
 * Matplotlib 3.3.2

--- a/astropy/visualization/wcsaxes/tests/test_misc.py
+++ b/astropy/visualization/wcsaxes/tests/test_misc.py
@@ -24,7 +24,9 @@ from astropy.visualization.wcsaxes.utils import get_coord_meta
 from astropy.visualization.wcsaxes.transforms import CurvedTransform
 
 mpl_version = Version(matplotlib.__version__)
+ft_version = Version(matplotlib.ft2font.__freetype_version__)
 MATPLOTLIB_LT_31 = mpl_version < Version('3.1')
+FREETYPE_261 = ft_version == Version("2.6.1")
 TEX_UNAVAILABLE = not matplotlib.checkdep_usetex(True)
 
 
@@ -504,13 +506,10 @@ def test_set_labels_with_coords(ignore_matplotlibrc, frame_class):
         assert ax.coords[i].get_axislabel() == labels[i]
 
 
-def test_bbox_size():
-    # Test for the size of a WCSAxes bbox
-    x0 = 11.38888888888889
-    y0 = 3.5
-    # Upper bounding box limits (only have Matplotlib >= 3.0 now)
-    x1 = 576.0
-    y1 = 432.0
+@pytest.mark.parametrize('atol', [0.2, 1.0e-8])
+def test_bbox_size(atol):
+    # Test for the size of a WCSAxes bbox (only have Matplotlib >= 3.0 now)
+    extents = [11.38888888888889, 3.5, 576.0, 432.0]
 
     fig = plt.figure()
     ax = WCSAxes(fig, [0.1, 0.1, 0.8, 0.8])
@@ -518,7 +517,8 @@ def test_bbox_size():
     fig.canvas.draw()
     renderer = fig.canvas.renderer
     ax_bbox = ax.get_tightbbox(renderer)
-    assert np.allclose(ax_bbox.x0, x0)
-    assert np.allclose(ax_bbox.x1, x1)
-    assert np.allclose(ax_bbox.y0, y0)
-    assert np.allclose(ax_bbox.y1, y1)
+
+    # Enforce strict test only with reference Freetype version
+    if atol < 0.1 and not FREETYPE_261:
+        pytest.xfail("Exact BoundingBox dimensions are only ensured with FreeType 2.6.1")
+    assert np.allclose(ax_bbox.extents, extents, atol=atol)

